pkg/processor/build/util/unarchive.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archiver/v4"
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+)
+
+type Unarchiver struct {
+	logger logger.Logger
+}
+
+func NewUnarchiver(parentLogger logger.Logger) (*Unarchiver, error) {
+	newUnarchiver := &Unarchiver{
+		logger: parentLogger,
+	}
+
+	return newUnarchiver, nil
+}
+
+func (d *Unarchiver) Extract(ctx context.Context, source string, target string) error {
+	if err := d.ExtractArchive(ctx, source, target); err != nil {
+		return errors.Wrapf(err, "Failed to extract archive %s", source)
+	}
+
+	return nil
+}
+
+func (d *Unarchiver) ExtractArchive(ctx context.Context, sourcePath string, targetPath string) error {
+	// open the source archive
+	file, err := os.Open(sourcePath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to open file")
+	}
+
+	// identify the type of the archive
+	format, input, err := archiver.Identify(sourcePath, file)
+	if err != nil {
+		return errors.Wrap(err, "Failed to identify archive")
+	}
+
+	handler := func(ctx context.Context, file archiver.File) error {
+		if filterFile, err := d.filterFile(file, targetPath); err != nil {
+			return errors.Wrap(err, "Failed to filter archived file")
+		} else if filterFile {
+			return nil
+		}
+
+		// copy the file to a new file with the same name in the target path
+		return d.extractFile(file, targetPath)
+	}
+
+	// want to extract something?
```

Comment:
Is it a question? :)

Reply:
leftover from my initial code 🙈 
will remove